ui/src/app/utils/index.tsx

```diff
@@ -329,7 +329,11 @@ export function sliceByHalfLength(obj) {
   return Math.ceil(obj.length / 2);
 }
 
-export function hasNewValidProps(currProps, prevProps, fieldsToCompare) {
+export function hasNewValidProps<
```

Comment:
functionality unchanged - just added types.  See its usage in admin-workspace for why I changed it here